### PR TITLE
Some minor linting, spelling, and code de-dup

### PR DESCRIPTION
**Pull request by nyurik**
Thanks for a great tool!  I used it a bit for my attempts at making Rust [a bit faster](https://github.com/rust-lang/rust/pull/121001) (no luck so far, but hopeful), so figured will contribute a bit back.

* introduced a new `Dumpable` trait and made MIR and LLVM implement it - so that `dump_function` can share both implementations (it was identical for both)
* removed a few un-needed/inconsistent qualifiers
* ran spellchecker
* inlined a few format args

### CODE REVIEW

**Comment by pacak** [comment]
If you run `cargo test` - it will update the readme file, adding that change should fix the test failure.

File: README.md [hunk 1]
```diff
@@ -193,7 +193,7 @@ Show the code rustc generates for any function
 
 
 
-You can start by running `cargo asm` with no parameters - it will suggests how to narrow the
```
Comment (pacak) [inline README.md:196]: those changes should go into `README.tpl`

Comment (nyurik) [reply README.md:196]: thx!

**Review by pacak — CHANGES_REQUESTED** [review]
Looks good other than quotes in the help message. I appreciate the documentation fixes :)

For the future - it's much easier to review similar requests if you separate typos, different type of changes and refactors into separate commits. So here it can be:
1. all the typos
2. fixing things like `println!("{}", x);`
3. fixing imports
4. `Dumpable` refactor.

File: src/opts.rs [hunk 2]
```diff
@@ -87,10 +87,10 @@ pub struct Cargo {
     /// Produce a build plan instead of actually building
     #[bpaf(hide_usage)]
     pub dry: bool,
-    /// Requires Cargo.lock and cache are up to date
+    /// Requires `Cargo.lock` and cache to be up-to-date
```
[the remark this answers is not in the document]
Comment (pacak) [reply src/opts.rs:90]: you need to update `opts.rs` too

**Review by pacak — APPROVED** [review]

**Comment by pacak** [comment]
Thank you for your contribution, they will go in with the next release that changes user facing bits :)

**Pull request merged**